tasks/broad/RNAWithUMIsTasks.wdl

```diff
@@ -0,0 +1,576 @@
+version 1.0
+
+task VerifyPipelineInputs {
+    meta {
+        description: "Verify that the pipeline input is either a ubam or pair of fastqs with additional info"
+    }
+
+    input {
+        File? bam
+        File? r1_fastq
+        File? r2_fastq
+
+        # fastq specific field
+        String? platform
+        String? library_name
+        String? platform_unit
+        String? read_group_name
+        String? sequencing_center = "BI"
+
+        String docker = "python:3.7.2"
+        Int cpu = 1
+        Int memory_mb = 1000
+        Int disk_size_gb = 1
+    }
+
+    command <<<
+        set -e
+        python3 <<CODE
+
+        fastq_flag = 0
+        bam = "~{bam}"
+        r1_fastq = "~{r1_fastq}"
+        r2_fastq = "~{r2_fastq}"
+        platform = "~{platform}"
+        library_name = "~{library_name}"
+        platform_unit = "~{platform_unit}"
+        read_group_name = "~{read_group_name}"
+        sequencing_center = "~{sequencing_center}"
+
+        if bam and not r1_fastq and not r2_fastq:
+            pass
+        elif r1_fastq and r2_fastq and not bam:
+            if platform and library_name and platform_unit and read_group_name and sequencing_center:
+                fastq_flag += 1
+        else:
+            raise ValueError("Invalid Input. Input must be either ubam or pair of fastqs with supplemental data")
+
+        with open("output.txt", "w") as f:
+            if fastq_flag == 1:
+                f.write("true")
+            # Remaining case is that only bam is defined
+            else: 
+                f.write("false")
+
+        CODE
+    >>>
+
+    runtime {
+        docker: docker
+        cpu: cpu
+        memory: "${memory_mb} MiB"
+        disks: "local-disk ${disk_size_gb} HDD"
+    }
+
+    output {
+        Boolean fastq_run = read_boolean("output.txt")
+    }
+}
+
+task ExtractUMIs {
+	input {
```

Comment:
I think this indent is a tab, but we usually use 4 spaces (This also goes for lines 72-74)